workspace/src/localization/localization_py/localization_py/particle_filter.py

```diff
@@ -0,0 +1,219 @@
+import numpy as np
+import math
+import matplotlib.pyplot as plt
+import random
+from scipy.stats import wasserstein_distance
+import yaml
+
+class particle_filter:
```

Comment:
```suggestion
class ParticleFilter:
```

See https://peps.python.org/pep-0008/#naming-conventions